src/lib/Libwin/pid.c

```diff
@@ -198,14 +197,14 @@ waitpid(HANDLE pid, int *statp, int opt)
 			if (ret == WAIT_TIMEOUT)
 				rval = 0;
 			else if (ret == WAIT_FAILED) { /* notion of an abnormal exit */
-				log_eventf(PBSEVENT_ERROR, 0, LOG_ERR, __func__, "WaitForSingleObject(%p, %d) failed with errno %d, i[%d]",
-					pid_handles[i], timeout, GetLastError(), i);
+				log_errf(-1, __func__, "WaitForSingleObject(%d, %d) failed i[%d/%d]", pid_handles[i], timeout, i, pids_cnt);
 				rval = (HANDLE)-1;
 				*statp = -1;
 			} else {
-				sprintf(logb,"found pid_handles[%d]=%p to have exited", i, pid_handles[i]);
+				sprintf(logb,"found pid_handles[%d]=%d to have exited", i, pid_handles[i]);
```

Comment:
Before this line should we check whether "ret == WAIT_OBJECT_0" ?

Reply:
as per the [doc](https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-waitforsingleobject#return-value) apart from `WAIT_OBJECT_0` the only other remaining return code remaining is `WAIT_ABANDONED`.  and `WAIT_ABANDONED` is only for mutex objects, and since we are not handling mutex handles here, we can safely ignore the check.   However if you still want to add the check let me know , i don't mind either ways

> 

WAIT_ABANDONED0x00000080L | The specified object is a mutex object that was not released by the thread that owned the mutex object before the owning thread terminated. Ownership of the mutex object is granted to the calling thread and the mutex state is set to nonsignaled.If the mutex was protecting persistent state information, you should check it for consistency.
-- | --

Reply:
Yes we are not handling any mutex here